addons/mail/static/src/web/messaging_menu/messaging_menu.js

```diff
@@ -53,20 +53,18 @@ export class MessagingMenu extends Component {
     }
 
     onClickPreview(isMarkAsRead, preview) {
-        const { thread } = preview;
         if (!isMarkAsRead) {
-            this.openDiscussion(thread);
+            this.openDiscussion(preview);
             return;
         }
         this.markAsRead(preview);
     }
 
     markAsRead(preview) {
-        const { thread, isNeedaction } = preview;
-        if (isNeedaction) {
-            this.threadService.markAllMessagesAsRead(thread);
+        if (preview.hasNeedactionMessages) {
+            this.threadService.markAllMessagesAsRead(preview);
         } else {
-            this.threadService.markAsRead(thread);
+            this.threadService.markAsRead(preview);
```

Comment:
This behavior needs to be different if it's a DM, both should be done or something like that.